embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,402 @@
+//! The polyline primitive
+
+use crate::{
+    draw_target::DrawTarget,
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point},
+    pixelcolor::PixelColor,
+    primitives::{
+        line::{self, Line},
+        Primitive, Rectangle,
+    },
+    style::{PrimitiveStyle, Styled},
+    transform::Transform,
+};
+
+/// Polyline primitive
+///
+/// Creates an unfilled chained line shape. Styles with a width greater than 1px are not currently
+/// supported and will always render as a 1px wide line.
+///
+/// # Examples
+///
+/// ## Draw a "heartbeat" shaped polyline
+///
+/// This example draws a stylized cardiogram in a 1px green stroke.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     pixelcolor::Rgb565, prelude::*, primitives::Polyline, style::PrimitiveStyle,
+/// };
+/// # use embedded_graphics::mock_display::MockDisplay;
+/// # let mut display = MockDisplay::default();
+///
+/// // A "heartbeat" shaped polyline
+/// let points: [Point; 10] = [
+///     Point::new(10, 64),
+///     Point::new(50, 64),
+///     Point::new(60, 44),
+///     Point::new(70, 64),
+///     Point::new(80, 64),
+///     Point::new(90, 74),
+///     Point::new(100, 10),
+///     Point::new(110, 84),
+///     Point::new(120, 64),
+///     Point::new(300, 64),
+/// ];
+///
+/// let line_style = PrimitiveStyle::with_stroke(Rgb565::GREEN, 1);
+///
+/// Polyline::new(&points)
+///     .into_styled(line_style)
+///     .draw(&mut display)?;
+/// # Ok::<(), core::convert::Infallible>(())
+/// ```
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub struct Polyline<'a> {
+    /// An offset to apply to the polyline as a whole
+    pub translate: Point,
+
+    /// All vertices in the line
+    pub vertices: &'a [Point],
+}
+
+impl<'a> Polyline<'a> {
+    /// Create a new polyline from a list of vertices
+    ///
+    /// If fewer than two vertices are provided, the line will not render anything when drawn.
+    pub fn new(vertices: &'a [Point]) -> Self {
+        Self {
+            vertices,
+            translate: Point::zero(),
+        }
+    }
+}
+
+impl<'a> Primitive for Polyline<'a> {
+    type PointsIter = Points<'a>;
+
+    fn points(&self) -> Self::PointsIter {
+        Points::new(self)
+    }
+}
+
+impl<'a> Dimensions for Polyline<'a> {
+    fn bounding_box(&self) -> Rectangle {
```

Comment:
I think this doesn't handle `vertices.len() < 2` correctly.
For `len = 0` the bounding box doesn't have a defined `top_left` corner, but should be `Size::zero()`.
For `len = 1` the correct behavior depends on how we decide to handle single vertex polylines (see comment above). Depending on that the size of the bounding box would be `Size::zero()` or `Size::new(1, 1)`.

Reply:
> For len = 0 the bounding box doesn't have a defined top_left corner

Hm. Should we just set it to `Point::zero()` in this case? It could also use the value of the `translate` field on `Polyline`.

Reply:
Yes, setting it to `Point::zero()` seems to be the best option IMO.